src/core/component.js

```diff
@@ -245,6 +251,12 @@ module.exports.registerComponent = function (name, definition) {
   var NewComponent;
   var proto = {};
 
+  if (name.indexOf('_') !== -1) {
```

Comment:
Are we preventing using single underscore too? Why not check for `indexOf('__')`